src/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as core from "@actions/core";
+import { read } from "@1password/op-js";
+import {
+	authErr,
+	envConnectHost,
+	envConnectToken,
+	envServiceAccountToken,
+	envManagedVariables,
+} from "./constants";
+
+export const semverToInt = (input: string): string =>
+	input
+		.split(".")
+		.map((n) => n.padStart(2, "0"))
+		.join("");
+
+export const validateAuth = (): void => {
+	let authType = "Connect";
+	if (!process.env[envConnectHost] || !process.env[envConnectToken]) {
+		if (!process.env[envServiceAccountToken]) {
+			throw new Error(authErr);
+		}
+		authType = "Service account";
+	}
+
+	// Adjust Connect host to have a protocol
+	if (
+		process.env[envConnectHost] &&
+		/* eslint-disable no-restricted-syntax */
+		!process.env[envConnectHost].startsWith("http://") &&
+		!process.env[envConnectHost].startsWith("https://")
+	) {
+		process.env[envConnectHost] = `http://${process.env[envConnectHost]}`;
```

Comment:
Pretty sure I already know the answer to this one, but just want to vocalize it here as a gut check: Connect host URLs are always `http` and not `https` because the Connect server is meant to be hosted inside of your own trusted network infrastructure - is that correct?

Reply:
Great context, thanks for expanding on the reasoning for this! I'm almost always in favour of encouraging people to be explicit with their expectations; asking users to specify either `http` or `https` as their protocol seems perfectly reasonable.